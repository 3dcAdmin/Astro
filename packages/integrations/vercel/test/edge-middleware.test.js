import { loadFixture } from './test-utils.js';
import { expect } from 'chai';

describe('Serverless prerender', () => {
	/** @type {import('./test-utils').Fixture} */
	let fixture;

	before(async () => {
		fixture = await loadFixture({
			root: './fixtures/middleware/',
			build: {
				excludeMiddleware: true,
			},
		});
	});

	it('build successfully the middleware edge file', async () => {
		await fixture.build();
		expect(
			await fixture.readFile(
				// this is abysmal...
				'../.vercel/output/functions/render.func/packages/integrations/vercel/test/fixtures/middleware/dist/middleware.mjs'
			)
		).to.be.ok;
	});
});
