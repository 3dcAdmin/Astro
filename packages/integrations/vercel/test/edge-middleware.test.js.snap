// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serverless prerender build successfully the middleware edge file 1`] = `
"// test/fixtures/middleware/dist/middleware2.mjs
var onRequest = async (context, next) => {
  const response = await next();
  return response;
};

// <stdin>
import { createContext, trySerializeLocals } from \\"astro/middleware\\";
async function middleware(request) {
  const url = new URL(request.url);
  const ctx = createContext({
    request,
    params: {}
  });
  const next = async () => {
    ctx.locals = (({ request: request2 }) => {
      console.log(request2);
      return {
        \\"foo\\": \\"bar\\"
      };
    })({ request });
    const response = await fetch(url, {
      headers: {
        \\"x-astro-locals\\": trySerializeLocals(ctx.locals)
      }
    });
    return response;
  };
  return onRequest(ctx, next);
}
export {
  middleware as default
};
"
`;
